lib/rtcpeerconnection/edge.js

```diff
@@ -76,6 +137,125 @@ EdgeRTCPeerConnection.prototype.close = function close() {
   });
 };
 
+// NOTE(syerrapragada): Rollback implementation copied from chrome shim
+EdgeRTCPeerConnection.prototype.addIceCandidate = function addIceCandidate(candidate) {
+  var self = this;
+  if (this.signalingState === 'have-remote-offer') {
+    return this._signalingStateLatch.when('low').then(function signalingStatesResolved() {
+      return self._peerConnection.addIceCandidate(candidate);
+    });
+  }
+  return this._peerConnection.addIceCandidate(candidate);
+};
+
+EdgeRTCPeerConnection.prototype.setLocalDescription = function setLocalDescription(description) {
+  return setDescription(this, true, description);
+};
+
+EdgeRTCPeerConnection.prototype.setRemoteDescription = function setRemoteDescription(description) {
+  return setDescription(this, false, description);
+};
+
+EdgeRTCPeerConnection.prototype.createAnswer = function createAnswer(options) {
+  var self = this;
+
+  if (this._pendingRemoteOffer) {
+    return this._peerConnection.setRemoteDescription(this._pendingRemoteOffer).then(function setRemoteDescriptionSucceeded() {
+      self._signalingStateLatch.lower();
+      return self._peerConnection.createAnswer();
+    }).then(function createAnswerSucceeded(answer) {
+      self._pendingRemoteOffer = null;
+      return answer;
+    }, function setRemoteDescriptionOrCreateAnswerFailed(error) {
+      self._pendingRemoteOffer = null;
+      throw error;
+    });
+  }
+
+  return this._peerConnection.createAnswer(options);
+};
+
+function setDescription(peerConnection, local, description) {
+  function setPendingLocalOffer(offer) {
+    if (local) {
+      peerConnection._pendingLocalOffer = offer;
+    } else {
+      peerConnection._pendingRemoteOffer = offer;
+    }
+  }
+
+  function clearPendingLocalOffer() {
+    if (local) {
+      // NOTE(syerrapragada): When we rollback current offer,
+      // We also need to reset mid and sdpMLineIndex for transceivers mutated
+      // in current negotiation.
+      peerConnection._nonNegotiatedTracks.forEach(function(track) {
+        var transceiver = peerConnection._transceivers.find(function(transceiver) {
+          return transceiver.track === track;
```

Comment:
@syerrapragada Can you add a note that `.track` is supposed to be in the `sender` or `receiver` and not in the `transceiver`?